data-prepper-plugins/log-http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/LogHTTPSource.java

```diff
@@ -0,0 +1,93 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.plugins.source.loghttp;
+
+import com.amazon.dataprepper.model.PluginType;
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.buffer.Buffer;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.model.source.Source;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+@DataPrepperPlugin(name = "log_http_source", type = PluginType.SOURCE)
```

Comment:
This name should not contain `_source`. It doesn't add any value to somebody configuring a pipeline.

Additionally, does this work beyond just logs? Perhaps we can call this plugin just `http`?

The current `file` source is a good example to follow for this plugin.

Reply:
Renamed main classes correspondingly except for LogHttpService since it is not yet expected to work under generic data.